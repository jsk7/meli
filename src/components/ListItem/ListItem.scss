@import 'src/styles/common';
@import 'src/styles/variables';


.ItemQuickView {
  height: 100%;
  color: $darkest;

  .ItemQuickImage {
    width: $product-thumbnail-size;
    height: $product-thumbnail-size;
    border-radius: $border-radius;
    padding: 0 $base-margin;
    object-fit: contain;
  }

  .description {
    position: absolute;
    display: inline-block;
    width: 78%;

    * {
      font-weight: 400;
      margin: 0;
    }

    .header {
      //display: inline-flex;
      position: relative;
      width: 100%;
      margin-bottom: $double-margin;
      margin-top: $base-margin;
    }

    .shipping {
      @include vertical-align;
      position: absolute;
      width: 24px;
      height: 24px;
      bottom: 0;
      margin-left: $base-margin;
    }

    .price {
      display: inline-flex;
      font-size: $medium-font;

      &.placeholder {
        width: 80px;
        &::after {
          content: "";
          margin-top: 5px;
          margin-left: 5px;
          display: inline-table;
          width: $medium-font*3;
          background-color: rgba(0, 0, 0, 0.2);
          border-radius: $border-radius;
          height: $medium-font;
        }
      }
    }
    .title {
      font-size: $small-font;
      line-height: 1.2em;
      margin: 0;

      &.placeholder {
        &::after {
          content: "";
          display: inline-table;
          width: $medium-font*10;
          background-color: rgba(0, 0, 0, 0.1);
          border-radius: $border-radius;
          height: $medium-font;
          @include placeholder-vertical-align;
        }
      }

      &.placeholderSubtitle {
        display: flex;
        &::after {
          content: "";
          display: inline-table;
          width: $medium-font*6;
          background-color: rgba(0, 0, 0, 0.05);
          border-radius: $border-radius;
          height: $medium-font;
          top: 0px;
          //@include placeholder-vertical-align;
        }
      }
    }

    .location {
      position: absolute;
      color: $dark;
      float: right;
      margin-right: 10%;
      bottom: 0;
      right: 0;
      vertical-align: bottom;
      font-size: $smallest-font;
      display: inline-flex;
    }
  }
}
