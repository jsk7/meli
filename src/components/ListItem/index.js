import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './ListItem.scss';

export default class ListItem extends React.Component {

  render() {
    const { item, lang } = this.props;
    const location = item.seller_address && item.seller_address.state && item.seller_address.state.name
    const priceClassname = classnames({
      price: true,
      placeholder: !item.price
    });
    const titleClassname = classnames({
      title: true,
      placeholder: !item.title
    })
    const subtitleClassname = classnames({
      title: true,
      placeholderSubtitle: !item.condition
    })
    return (
      <div className="ListItem" >
        <div className="image-helper">
          <img src={item.thumbnail} className="ItemQuickImage" />
        </div>
        <div className="description col-9 col-7-sm">
          <div className="header">
            <h2 className={priceClassname}>{`$ ${item.price}`}</h2>
            <img className="shipping"
                  src={require('../../assets/ic_shipping@2x.png.png')}
            />
            <h4 className="location">{location}</h4>
          </div>
          <h3 className={titleClassname}>
            <span className="capitalize">
              {item.title}
            </span>
          </h3>
          <h3 className={subtitleClassname}>
            <span className="capitalize">
              {lang[item.condition]}
            </span>
          </h3>
        </div>
      </div>
    )
  }
}

ListItem.propTypes = {
  item: PropTypes.object.isRequired,
  lang: PropTypes.object.isRequired
}
