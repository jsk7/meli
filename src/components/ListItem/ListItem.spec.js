import React from 'react';
import ListItem from './index';
import renderer from 'react-test-renderer';
import itemsMock from '../../mocks/items';

describe('<ListItem />', () => {
  it('should render correctly if it has an item', () => {
    const tree = renderer.create(
      <ListItem item={itemsMock[0]} lang={{}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render placeholders if it has no item', () => {
    const tree = renderer.create(
      <ListItem item={{}} lang={{}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
