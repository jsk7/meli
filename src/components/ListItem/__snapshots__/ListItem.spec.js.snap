// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render correctly if it has an item 1`] = `
<div
  className="ListItem"
>
  <div
    className="image-helper"
  >
    <img
      className="ItemQuickImage"
      src="http://mla-s2-p.mlstatic.com/893911-MLA25840454871_082017-I.jpg"
    />
  </div>
  <div
    className="description col-9 col-7-sm"
  >
    <div
      className="header"
    >
      <h2
        className="price"
      >
        $ 7500
      </h2>
      <img
        className="shipping"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <h4
        className="location"
      >
        Capital Federal
      </h4>
    </div>
    <h3
      className="title"
    >
      <span
        className="capitalize"
      >
        Apple Ipod Touch 32gb 6ta Generacion
      </span>
    </h3>
    <h3
      className="title"
    >
      <span
        className="capitalize"
      />
    </h3>
  </div>
</div>
`;

exports[`<ListItem /> should render placeholders if it has no item 1`] = `
<div
  className="ListItem"
>
  <div
    className="image-helper"
  >
    <img
      className="ItemQuickImage"
      src={undefined}
    />
  </div>
  <div
    className="description col-9 col-7-sm"
  >
    <div
      className="header"
    >
      <h2
        className="price placeholder"
      >
        $ undefined
      </h2>
      <img
        className="shipping"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <h4
        className="location"
      />
    </div>
    <h3
      className="title placeholder"
    >
      <span
        className="capitalize"
      />
    </h3>
    <h3
      className="title placeholderSubtitle"
    >
      <span
        className="capitalize"
      />
    </h3>
  </div>
</div>
`;
